sdk/core/core-http-compat/README.md

```diff
@@ -0,0 +1,64 @@
+# Azure Core HTTP Compatibility library for JavaScript
+
+This library provides compatibility between the Core V1 ([core-http](https://www.npmjs.com/package/@azure/core-http)) and the Core V1 ([core-client](https://www.npmjs.com/package/@azure/core-client) & [core-rest-pipeline](https://www.npmjs.com/package/@azure/core-rest-pipeline)) libraries.
+
+Recently, a new set of libraries (`core-client` & `core-rest-pipeline`) have been developed to replace the older `core-http` library. The existing libraries with `core-http` dependency will be migrated to use the new libraries. But, this migration will cause some breaking changes.
+
+In order to avoid the breaking change to our customers, this compatibility library has been developed. This will avoid any breaking changes during the migration and provide a bridging layer between Core V1 & Core V2 libraries.
+
+## Usage
+
+### ExtendedServiceClientOptions
+
+With Core V1 library, the `options` parameter to the generated client will look like:
+
+```
+export interface SearchClientOptionalParams extends coreHttp.ServiceClientOptions {
+  /** Overrides client endpoint. */
+  endpoint?: string;
+}
```

Comment:
What do you think of combining all the code related to core v1 in a single code snippet? And then another for core v2? Then we will be showing options and how to use the options all in a single place. Is there a case where someone would want need to understand only the options and not creating the client? Or only the ExtendedServiceClientOptions and not ExtendedCommonClientOptions?

Reply:
Actually Yes. One is for generated client and one is for custom client. That was the reason I am showing them seperately.

Reply:
The changes to the generated client will be taken care by the code generator and the end user does not have to do anything right? So, am thinking we don;t need to go into too much details here or we can skip the section for generated client altogether

> If your library was auto generated using the `@autorest/typescript` code generator and it added a dependency on `@azure/core-http`